experiments/compositions/composition/internal/controller/expander_reconciler.go

```diff
@@ -584,6 +589,26 @@ func (r *ExpanderReconciler) evaluateAndSavePlan(ctx context.Context, logger log
 	return values, updated, "", nil
 }
 
+func (r *ExpanderReconciler) enqueueAllFromGVK(ctx context.Context, _ client.Object) []reconcile.Request {
+	logger := log.FromContext(ctx)
+	logger.Info("Got notification of changed CRD")
+	inputcrList := &unstructured.UnstructuredList{}
+	inputcrList.SetGroupVersionKind(r.InputGVK)
+	if err := r.List(ctx, inputcrList); err != nil {
+		logger.Error(err, "unable to fetch Input API Objects")
+		return nil
+	}
+	if len(inputcrList.Items) == 0 {
+		return nil
+	}
+	var reqs []reconcile.Request
+	for _, inputcr := range inputcrList.Items {
```

Comment:
This would cause thundering herd and thats fine. Wondering if the controller Q can handle this for large numbers. Please add a comment to that effect so that we know this is a place where we could optimize later when needed.

Reply:
Will do.